File: src/luogu/questions/p1115/Main.cpp
// https://www.luogu.com.cn/problem/P1115
#include <bits/stdc++.h>
using namespace std;
// TODO: 1WA, 3TLE
int main() {
  int n;
  cin >> n;
  int array[n + 1];
  int tmp;
  for (int i = 1; i < n + 1; i++) {
    cin >> tmp;
    array[i] = array[i - 1] + tmp;
  }
  int max = 0;
  for (int i = 0; i < n + 1; i++) {
    for (int j = i + 1; j < n + 1; j++) {
      if (array[j] - array[i] > max)
        max = array[j] - array[i];
    }
  }
  cout << max << endl;
  return 0;
}

File: src/luogu/questions/p1115/Main.java
// https://www.luogu.com.cn/problem/P1115
package luogu.questions.p1115;

import java.util.Scanner;

// TODO: 1WA, 3TLE
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] arr = new int[n + 1];
        for (int i = 1; i <= n; i++) {
            arr[i] = arr[i - 1] + sc.nextInt();
        }
        // System.out.println(Arrays.toString(arr));
        int max = 0;
        for (int i = 0; i < n + 1; i++) {
            for (int j = i + 1; j < n + 1; j++) {
                if (arr[j] - arr[i] > max)
                    max = arr[j] - arr[i];
            }
        }
        System.out.println(max);
    }
}
